NitroxPatcher/Patches/SeaMoth_OnUpgradeModuleUse_Patch.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Reflection;
+using Harmony;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxModel.Core;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches
+{
+    public class SeaMoth_OnUpgradeModuleUse_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(SeaMoth);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("OnUpgradeModuleUse", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        public static bool Prefix(SeaMoth __instance, TechType techType, int slotID)
+        {
+            NitroxServiceLocator.LocateService<SeamothModulesEvent>().Update(techType, slotID, __instance);
+
+            using (NitroxServiceLocator.LocateService<IPacketSender>().Suppress<ItemContainerRemove>())
+            {
+                //Need Suppress ItemContainerRemove Generate for this Function
+                bool flag = true;
+                float num = 0f;
+                if (techType != TechType.SeamothSonarModule)
+                {
+                    if (techType != TechType.SeamothElectricalDefense)
+                    {
+                        if (techType == TechType.SeamothTorpedoModule)
```

Comment:
These 'if' statements are very nested.  Can we change to:

if(techType == TechType.SeamonthSonar) {
    sonarActivated();
} elseif(techType == TechType.SeamothElectricalDefense) {
    electricalDefenseActivated();
} else if(techType == TechType.SeamothTorpedoModule) {
    torpedoActivated();
}